File: data/Dtos/DeliveryPlan.cs
using data.Entities;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using data.Utility.Paging;

namespace data.Dtos
{
    public class DeliveryPlanRes
    {
        public int DeliveryPlanId { get; set; }
        public int CourierId { get; set; }
        public Courier? Courier { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public decimal Fee { get; set; }
    }

    public class DeliveryPlanReqEdit
    {
        [Required]
        public int CourierId { get; set; }
        [Required, MaxLength(50)]
        public string? Name { get; set; }
        public string? Description { get; set; }
        public decimal Fee { get; set; }
    }

    public class DeliveryPlanReqSearch : PagedRequestDto
    {
        [Required]
        public int CourierId { get; set; }
    }
}

File: web/src/dtos/DeliveryPlan.ts
import Common from "../utility/Common";
import { CourierRes } from "./Courier";
import { PagedReq } from "./PagedReq";

export interface DeliveryPlanRes {
  deliveryPlanId?: number;
  courierId?: number;
  courier?: CourierRes;
  name?: string;
  description?: string;
  fee?: number;
}

export class DeliveryPlanReqEdit {
  courierId?: number = 0;
  name?: string = "";
  description?: string = "";
  fee?: number = 0;

  constructor(courierId: number) {
    this.courierId = courierId;
  }
}

export class DeliveryPlanReqSearch extends PagedReq {
  courierId?: number;

  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {courierId = 0}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.courierId = courierId;
  }
}